editor/editor_inspector.cpp

```diff
@@ -3240,6 +3232,14 @@ void EditorInspector::_edit_set(const String &p_name, const Variant &p_value, bo
 		Variant value = object->get(p_name, &valid);
 		if (valid) {
 			undo_redo->add_undo_property(object, p_name, value);
+
+			for (EditorProperty *E : editor_property_map[p_name]) {
+				if (E->is_checkable()) { // Properties like theme overrides.
+					// Not null (nil or nullptr) means overridden, the state of the checkbox is checked, that is true.
+					undo_redo->add_do_method(E, "set_checked", !(p_value.get_type() == Variant::NIL || (p_value.get_type() == Variant::OBJECT && p_value.is_null())));
+					undo_redo->add_undo_method(E, "set_checked", !(value.get_type() == Variant::NIL || (value.get_type() == Variant::OBJECT && value.is_null())));
```

Comment:
Could `E->is_checked()` be used here?

Reply:
It only can be used in `add_undo_method` .

Reply:
There are cases where continuously changing the value does not change the checkbox state.

Reply:
So the value used by `add_do_method` should be determined according to the set value.